shared-bin/src/catchers.rs

```diff
@@ -0,0 +1,21 @@
+use http_api_problem::HttpApiProblem;
+use http_api_problem::StatusCode;
+use rocket::Request;
+use rocket_basicauth::unauthorized;
+
+/// Emit HttpApiProblem whenever a Unprocessable Entity error (422) happens
+///
+/// Rocket emits this error when deserialised JSON is syntactically valid but
+/// it's not valid *semantically* (e.g. when passed 0 to a NonZerou32).
+/// Display this as error 400 as 422 is Web-Dav specific.
+#[rocket::catch(422)]
+pub fn unprocessable_entity(req: &Request) -> HttpApiProblem {
+    HttpApiProblem::new(StatusCode::BAD_REQUEST)
+        .title("Bad Request")
+        .detail(format!("{}", req.uri()))
```

Comment:
Fundamentally, the client should know which URI it was hitting.

If we want to display the URI in the error message, I'd suggest we add that client side or leave it out altogether.

Perhaps we can say "Invalid JSON" or something?

Reply:
As a user, it was helpful to see the "/api/offer" there. Eventually I'd like to highlight the problem in the particular field, but we don't have/support field validation at this stage unfortunately. 

Not sure whether 'Invalid JSON' would be better for the clients, showing this in the UI would confuse the users even more?